### ISSUE TITLE
Unflat files will loose keys on special conditions

### ISSUE THREAD

**Issue by darkv** [ngneat/transloco-keys-manager#23]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Having some translation keys like

- top.middle
- top.middle.last1
- top.middle.last2

will create three corresponding entries in the translation files:

```
{
  "top.middle" : "a";
  "top.middle.last1" : "b";
  "top.middle.last2" : "c";
}
```

If you chose to use unflat files by setting the config to `unflat: true` this will be converted either to

```
{
  "top": {
    "middle": "a"
  }
}
```

or

```
{
  "top": {
    "middle": {
      "last1": "b",
      "last2": "c"
    }
  }
}
```

depending on the order of processing the keys and thus leading to a loss of information. This behavior is mainly related to the used lib _flat_. So in the current implementation the feature "unflat" can be very dangerous as the user won't notice that loss of keys because there is neither an error/warning output during key extraction nor a warning in the documentation.

## Expected behavior

It has to be decided if the current behavior can be fixed by either trying to get a fix on the flat lib or by introducing some sort of preprocessor for unflat files to amend the keys so that flat will behave correctly. That preprocessor would detect such cases and modify a list of extracted keys like

```
{
  …
  "top.middle" : "a",
  "top.middle.last1" : "b",
  "top.middle.last2" : "c",
  …
}
```

to

```
{
  …
  "top.middle._" : "a",
  "top.middle.last1" : "b",
  "top.middle.last2" : "c",
  …
}
```

Then during reading those files back in the reverse processing has to be applied. This is just some sketchy idea and is meant as start of discussion on that topic.

**Comment by shaharkazaz** [comment]
@darkv this doesn't make sense to me... if you want to support the translations as objects `top.middle` can't wear both hats.

**Comment by darkv** [comment]
Sorry if I was writing too confusing. With the default configuration of transloco the generated translation files will be flat. So having keys like in the example above is no problem as every key will create just one additional line within the translation file(s).

So in the current behavior when switching to unflat files—which is very helpful as this enables collapsing subtrees in most of text editors automagically—the problem arises that some keys will just vanish unnoticed. A common use case of such kind of keys would be e.g. enum classes where you have a general key for a localized label for the enum itself and subkeys for the different enum values.

The problem I want to point out is that the current documentation lacks a warning that you have to account for this restriction and be sure to check for such cases yourself, because the keys manager just silently converts to unflat without any hint. This is ok if you have only a small amount of keys but is a difficult task if you are using transloco in bigger applications with hundreds or thousands of keys, many parallel working people, …

That's why I wanted to raise this point as to discuss if there should be some thought on how to fix this issue to make unflat a hassle-free config option or alternatively add appropriate information to the documentation and output during keys manager execution respectively.

**Comment by shaharkazaz** [comment]
@darkv I think that we will add appropriate information to the documentation. would you like to make the PR?

**Comment by shaharkazaz** [comment]
Published in v2.0.3, @darkv great job and thanks for the contribution 🚀
